### ISSUE TITLE
Associated nodes are not present in before_create callback

### ISSUE THREAD

**Issue by jmdeldin** [neo4jrb/neo4j#1043]
Hello,

I'm finding that my associations are unavailable to me in a `before_(save|create)` callback. Is there something I'm doing wrong in the following example?

```ruby
class TestUser
  include Neo4j::ActiveNode

  property :email

  has_many :out, :test_devices, type: :DEVICE_FOR
end

class TestDevice
  include Neo4j::ActiveNode

  property :token

  has_one :in, :test_user, type: :DEVICE_FOR

  before_create do
    puts "no user!" if test_user.nil?
  end
end

TestUser.destroy_all
TestDevice.destroy_all

u = TestUser.create!(email: 'foo@example.com')
d = TestDevice.new(token: 'sekret')
d.test_user = u
d.save!
```

For me under v5.2.14 and v6.0.0.alpha12, this prints "no user!".

**Comment by subvertallchris** [comment]
Someone also reported something similar in the Gitter room just now. We'll look into it and try to have a fix soon. Happy to take a PR if you dive in and fix it before us.

**Comment by cheerfulstoic** [comment]
I took a bit of time to look at this and I think it's because we're getting two different association_proxy objects.  One of them is storing the deferred object and one is getting generated freshly without it.

Maybe we should be storing the deferred objects for associations directly on the object?  Maybe just in a hash with the association name symbol as the keys?

**Comment by cheerfulstoic** [comment]
@jmdeldin I think I've got something that might fix it.  Could you try this in your Gemfile?

```ruby
gem 'neo4j', github: 'neo4jrb/neo4j', branch: 'unpersisted_association_refactor'`
```

That's a branch off of 6.0.0, by the way.

**Comment by jmdeldin** [comment]
@cheerfulstoic Excellent! That works for me in my test case. Thanks so much for fixing!

**Comment by cheerfulstoic** [comment]
Wonderful, thanks.  Just released 6.0.0 with this fix!  Let me know if there are still any problems
